provisioning/roles/rebooting_tasks/vars/main.yml

```diff
@@ -24,6 +23,14 @@ _kernel_packages:
     - kernel-core
     - kernel-modules
 
+  # We need to add kernel package for Fedora before Fedora 41,
+  # CentOS and RHEL
+  - Fedora([1-9][0-9]{2,}|2[8-9]|3[0-9]|40):
```

Comment:
The "[1-9][0-9]{2,}" part is for matching far-future version numbers with three digits. Since we're assuming the need for the "kernel" package stops with Fedora 40, it should be removed.